systemtests/src/main/java/com/github/streamshub/systemtests/utils/resourceutils/KafkaTopicUtils.java

```diff
@@ -0,0 +1,201 @@
+package com.github.streamshub.systemtests.utils.resourceutils;
+
+import com.github.streamshub.systemtests.clients.KafkaClients;
+import com.github.streamshub.systemtests.constants.Constants;
+import com.github.streamshub.systemtests.constants.Labels;
+import com.github.streamshub.systemtests.logs.LogWrapper;
+import com.github.streamshub.systemtests.utils.WaitUtils;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.skodjob.testframe.resources.KubeResourceManager;
+import io.strimzi.api.ResourceAnnotations;
+import io.strimzi.api.ResourceLabels;
+import io.strimzi.api.kafka.model.nodepool.KafkaNodePool;
+import io.strimzi.api.kafka.model.nodepool.KafkaNodePoolBuilder;
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+import io.strimzi.api.kafka.model.topic.KafkaTopicBuilder;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class KafkaTopicUtils {
+    private static final Logger LOGGER = LogWrapper.getLogger(KafkaTopicUtils.class);
+    
+    private KafkaTopicUtils() {}
+    
+    public static List<KafkaTopic> createTopicsAndReturn(String namespace, String kafkaName, String topicNamePrefix, int numberToCreate, boolean waitForTopics, int partitions, int replicas, int minIsr) {
+        LOGGER.info("Create {} topics for cluster {} with topic name prefix {}", numberToCreate, kafkaName, topicNamePrefix);
+
+        List<KafkaTopic> topics = IntStream.range(0, numberToCreate)
+            .mapToObj(i -> defaultTopic(namespace, kafkaName, topicNamePrefix + "-" + i, partitions, replicas, minIsr).build())
+            .toList();
+
+        if (waitForTopics) {
+            topics.forEach(KubeResourceManager.get()::createResourceWithWait);
+        } else {
+            topics.forEach(KubeResourceManager.get()::createResourceWithoutWait);
+        }
+        return topics;
+    }
+
+
+    public static List<KafkaTopic> createUnderReplicatedTopicsAndReturn(String namespace, String kafkaName, String kafkaUser, String topicNamePrefix, int numberToCreate, int messageCount, int partitions, int replicas, int minIsr) {
```

Comment:
That's a bit strange name for the actions it really does.
With `createUnderReplicatedTopicsAndReturn` I would imagine that it will create just the KafkaTopic CRs with some strange configuration. But you are scaling the nodePools up, waiting for scale up, then you are creating topics, sending messages  and then scaling down.

How many times you will use this? Shouldn't it be split into multiple methods?

Reply:
It's used multiple times for a simple creation of udnerreplicated topics, i think you cannot create underreplicated topic with plain CR so the name made perfect sense for me it does create those topics and returns them for later usage.... maybe a more fiting name starting with `prepare` or `provision` maybe even `setup` ? i mean i do not want to repeat the whole scaling and moving partitions each test where it's used it would inflate all the tests method block

Reply:
That's what I mean, it does a lot of other things that the name actually suggests. Plus, you are missing the Javadoc there, which would help also others to understand what is the method doing.
And if you have multiple tests checking the things with same setup of topics - in this case under replicated - I think you can have it in some before all and then run against it few tests? Because this will take a lot of time if you will have it in every test.